### PR TITLE
Improve GlobalWindow, IntervalWindow Coders

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
Implement consistentWithEquals and verifyDeterministic on both of these
coders.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @dhalperi

**Comment by tgroh** [comment]
Required for good performance of Structural Values for https://github.com/apache/beam/pull/2626

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/IntervalWindow.java [hunk 1]
```diff
@@ -195,6 +195,17 @@ public IntervalWindow decode(InputStream inStream, Context context)
...
 
     @Override
+    public void verifyDeterministic() throws NonDeterministicException {
+      instantCoder.verifyDeterministic();
+      durationCoder.verifyDeterministic();
+    }
+
+    @Override
+    public boolean consistentWithEquals() {
+      return true;
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/IntervalWindow.java]: shouldn't this be a function of the component coders?

Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/IntervalWindow.java]: Yes. Done.

**Comment by dhalperi** [comment]
LGTM, please self merge upon green.

**Pull request closed without merging**